### PR TITLE
[3.0] backport Skip unready nodes deployment

### PR DESCRIPTION
**Pull request by ilausuch**
Adds to the experimental options the skip_unready_nodes
configs that would allow to set a list of roles for which we would not
have to wait for all nodes to be in status ready to apply the barclamps.

Backport of https://github.com/crowbar/crowbar-core/pull/1361

### CODE REVIEW

**Comment by Itxaka** [comment]
This is failing due the experimental options commit not being backported to cloud 6 so the `Rails.application.config.experimental` in here https://github.com/crowbar/crowbar-core/pull/1387/files#diff-0be8ea1f05c9441a59bb2b7ac65b1c62R941 fails to load.

This should be backported as part of this PR: https://github.com/crowbar/crowbar-core/commit/6c6b2ce42d529be3e708ed5155008b6f8b45fd06 but modified to remove the `disallow_restart` flag? @vuntz ?

**Comment by Itxaka** [comment]
[Backport "do not require all nodes to be ready" patch to SOC6CD](https://trello.com/c/3Ae6H2o4/177-backport-do-not-require-all-nodes-to-be-ready-patch-to-soc6cd)

File: crowbar_framework/app/models/service_object.rb [hunk 1]
```diff
@@ -960,33 +967,56 @@ def apply_role(role, inst, in_queue)
...
+    unless old_deployment.nil?
+      old_elements = old_deployment["elements_expanded"]
+      if old_elements.nil?
+        old_elements = old_deployment["elements"]
+      end
     end
 
-    @logger.debug "delay empty - running proposal"
+    # When bootstrapping, we don't run chef, so there's no need for queuing
+    if bootstrap
```
Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: CI fails: `Failed to talk to service proposal commit: 405: {"error":"Failed to apply the proposal: uncaught exception (undefined local variable or method `bootstrap' for #\u003cCrowbarService:0x007f48801966b0\u003e)"}`

File: crowbar_framework/config/experimental.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+default: &default
+  disallow_restart:
+    enabled: false
```
Comment (vuntz) [inline crowbar_framework/config/experimental.yml]: I probably wouldn't include this one, as the option doesn't exist here.

File: crowbar_framework/app/models/service_object.rb [hunk 3]
```diff
@@ -1659,11 +1695,33 @@ def active_update(proposal, inst, in_queue)
     end
   end
 
-  def only_if_admin(node)
-    yield if node.admin?
-  end
-
-  def only_unless_admin(node)
-    yield unless node.admin?
```
Comment (vuntz) [inline crowbar_framework/app/models/service_object.rb]: I don't believe you want to remove this.

Comment (ilausuch) [reply crowbar_framework/app/models/service_object.rb]: Yes, this seems a mistake, I'll fix it

**Comment by ilausuch** [comment]
I close this PR because I want to send it again with the commit history fixed

**Pull request closed without merging**